beacon/validator/src/main/java/tech/pegasys/teku/validator/coordinator/BlockOperationSelectorFactory.java

```diff
@@ -218,21 +218,20 @@ private SafeFuture<Void> setExecutionData(
               blockSlotState.getSlot()));
     }
 
-    // if requestedBlinded has been specified, we strictly follow it, otherwise we should run
-    // Builder flow (blinded) only if we have a validator registration
+    // We should run Builder flow (blinded) only if we have a validator registration
     final boolean shouldTryBuilderFlow =
-                executionPayloadContext
-                    .map(ExecutionPayloadContext::isValidatorRegistrationPresent)
-                    .orElse(false);
+        executionPayloadContext
+            .map(ExecutionPayloadContext::isValidatorRegistrationPresent)
+            .orElse(false);
 
     // pre-Merge Execution Payload / Execution Payload Header
     if (executionPayloadContext.isEmpty()) {
-      if (shouldTryBuilderFlow) {
```

Comment:
Hmm why did we remove the if statement?

Reply:
if `executionPayloadContext` is empty then `shouldTryBuilderFlow` will always be false

Reply:
Oh yes, got it. In this case, we can just leave the `.executionPayload`

Reply:
And move shouldTryBuilderFlow initialization afterwards

Reply:
leaving just the `.executionPayload` break one of the bellatrix unit tests (`shouldIncludeExecutionPayloadHeaderOfDefaultPayload`). which in my opinion should probably be fine to be removed but I'd like a second opinion here.

Reply:
Yeah, we can just remove that test, it is no longer applicable. We will always use the unblinded flow in that case.

Reply:
I think leaving only .executionPayload() will fix the acceptance test failure